### PR TITLE
feat(eigen-client-m0-implementation): grafana metrics

### PR DESCRIPTION
**Pull request by juan518munoz**
## What ❔

Implement grafana metrics for data availability dispatcher

image

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zkstack dev fmt` and `zkstack dev lint`.

### CODE REVIEW

File: core/node/da_clients/src/eigen/eigenda-integration.md [hunk 1]
```diff
@@ -113,6 +113,34 @@ You may enable observability here if you want to.
...
+(around line 108)
+
+Then modify the `era-observability/etc/prometheus/prometheus.yml` with the retrieved port:
+
+```yaml
+scrape_interval: 5s
+honor_labels: true
+static_configs:
+  - targets: ['host.docker.internal:3312'] # <- change this to the port
+```
```
Comment (gianbelinche) [inline core/node/da_clients/src/eigen/eigenda-integration.md:134]: We should indicate which one exactly to change this, it should be the one named zksync

File: core/node/da_clients/src/eigen/eigenda-integration.md [hunk 2]
```diff
@@ -113,6 +113,34 @@ You may enable observability here if you want to.
...
+
+2. Enable the Data Availability Grafana dashboard
+
+```bash
+mv era-observability/additional_dashboards/EigenDA.json era-observability/dashboards/EigenDA.json
+```
+
+3. Restart the era-observability container
```
Comment (gianbelinche) [inline core/node/da_clients/src/eigen/eigenda-integration.md:142]: Maybe say how to do it

File: zkstack_cli/crates/zkstack/src/commands/ecosystem/setup_observability.rs [hunk 3]
```diff
@@ -20,7 +20,16 @@ pub fn run(shell: &Shell) -> anyhow::Result<()> {
...
+    // Add lambda remote and checkout to `eigenda` for DA metrics
+    git::add_remote(
+        shell,
+        path_to_era_observability.clone(),
+        "lambda",
+        "https://github.com/lambdaclass/era-observability.git",
+    )?;
+    git::checkout(shell, path_to_era_observability.clone(), "eigenda")?;
+
+    spinner.finish();
```
Comment (gianbelinche) [inline zkstack_cli/crates/zkstack/src/commands/ecosystem/setup_observability.rs:33]: I think the idea is to not do this, that is why we have a PR to era-observability

**Review by gianbelinche — CHANGES_REQUESTED** [review]

File: core/node/da_clients/src/eigen/eigenda-integration.md (hunk 2 above)
Comment (juan518munoz) [reply core/node/da_clients/src/eigen/eigenda-integration.md:143]: I purposedly didn't add it as it's just two button click process via the docker GUI, but on second thought I've added a bash command to restart the containers.

File: zkstack_cli/crates/zkstack/src/commands/ecosystem/setup_observability.rs (hunk 3 above)
Comment (juan518munoz) [reply zkstack_cli/crates/zkstack/src/commands/ecosystem/setup_observability.rs:33]: If that's the case then an temporary step should be added to the instructions, specifying how to checkout to the needed branch while said PR remains unmerged.

Comment (gianbelinche) [reply zkstack_cli/crates/zkstack/src/commands/ecosystem/setup_observability.rs:33]: Let's add that and remove that code then!

Comment (juan518munoz) [reply zkstack_cli/crates/zkstack/src/commands/ecosystem/setup_observability.rs]: Done!

**Review by gianbelinche — APPROVED** [review]
LGTM

**Review by juanbono — APPROVED** [review]

**Pull request merged**